### PR TITLE
Add a script for crio installation for node e2e

### PR DESCRIPTION
**Pull request by harche**
Signed-off-by: Harshal Patil <harpatil@redhat.com>

#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
Custom installation script for crio for node e2e tests

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by harche** [comment]
The corresponding ignition file would look like, 

```
{
  "ignition": {
    "version": "3.1.0"
  },
  "systemd": {
    "units": [
      {
        "contents": "[Unit]\nDescription=Enable cgroup v1\nBefore=network-online.target\nConditionFirstBoot=yes\n\n[Service]\nType=oneshot\nExecStart=/usr/bin/rpm-ostree kargs --append systemd.unified_cgroup_hierarchy=0 --reboot\n[Install]\n\nWantedBy=multi-user.target\n",
        "enabled": true,
        "name": "cgroupv1.service"
      },
      {
        "contents": "[Unit]\nDescription=Download and install crio binaries and configurations.\nAfter=network-online.target\n\n[Service]\nType=oneshot\nExecStartPre=/usr/bin/bash -c '/usr/bin/curl --fail --retry 5 --retry-delay 3 --silent --show-error -o /usr/local/crio-nodee2e-installer.sh  https://storage.googleapis.com/node_e2e_crio_installer/node_e2e_installer?ignoreCache=1'\nExecStart=/usr/bin/bash /usr/local/crio-nodee2e-installer.sh\n\n[Install]\nWantedBy=multi-user.target\n",
        "enabled": true,
        "name": "crio-install.service"
      }
    ]
  }
}

```

File: scripts/node_e2e_installer [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /usr/sbin/setenforce 1
+
+    # Get the SELinux package
+    /usr/bin/curl --fail --retry 5 --retry-delay 3 --silent --show-error -o /tmp/kubelet-e2e.pp https://storage.googleapis.com/cri-o/selinux/kubelet-e2e.pp
+    /usr/sbin/semodule -i /tmp/kubelet-e2e.pp
+}
+
+install_crio () {
+    # Download and install CRIO
+    /usr/bin/curl --fail --retry 5 --retry-delay 3 --silent --show-error -o /usr/local/crio-install.sh  https://raw.githubusercontent.com/cri-o/cri-o/master/scripts/get
```
Comment (saschagrunert) [inline scripts/node_e2e_installer:17]: Is't `$PATH` set correctly during script execution that we could just
```suggestion
    curl --fail --retry 5 --retry-delay 3 --silent --show-error -o /usr/local/crio-install.sh  https://raw.githubusercontent.com/cri-o/cri-o/master/scripts/get
```

Comment (harche) [reply scripts/node_e2e_installer]: Thanks, updated.

**Comment by harche** [comment]
Removing WIP because has a successful run of node e2e in local setup.

**Comment by harche** [comment]
/hold 

Will verify cgroup v2 jobs as well.

**Review by saschagrunert — APPROVED** [review]
LGTM

**Comment by harche** [comment]
> /hold
> 
> Will verify cgroup v2 jobs as well.

Works with cgroup v2 job. 

/hold cancel

**Comment by saschagrunert** [comment]
Let's shfmt the script: https://github.com/cri-o/cri-o/pull/5167/checks?check_run_id=3250448612

**Comment by harche** [comment]
> Let's shfmt the script: https://github.com/cri-o/cri-o/pull/5167/checks?check_run_id=3250448612

Thanks, done.

File: scripts/node_e2e_installer [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    # Get the SELinux package
+    curl --fail --retry 5 --retry-delay 3 --silent --show-error -o /tmp/kubelet-e2e.pp https://storage.googleapis.com/cri-o/selinux/kubelet-e2e.pp
+    semodule -i /tmp/kubelet-e2e.pp
+}
+
+install_crio () {
+    # Download and install CRIO
+    curl --fail --retry 5 --retry-delay 3 --silent --show-error -o /usr/local/crio-install.sh https://raw.githubusercontent.com/cri-o/cri-o/master/scripts/get
+    bash /usr/local/crio-install.sh -t $NODE_E2E_COMMIT
```
Comment (haircommander) [inline scripts/node_e2e_installer:18]: nit: can we quote NODE_E2E_COMMIT

**Comment by haircommander** [comment]
@harche thanks, in general LGTM, besides a nit, and the fact that shfmt isn't happy yet...

File: scripts/node_e2e_installer (hunk 2 above)
Comment (harche) [reply scripts/node_e2e_installer]: @haircommander thanks, updated.

**Comment by haircommander** [comment]
still some shfmt failures:
https://github.com/cri-o/cri-o/pull/5167/checks?check_run_id=3254057817

File: scripts/node_e2e_installer [hunk 3]
```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+NODE_E2E_COMMIT=4d14fd61d2d533a0bb81557b54d3505be830efa7  # Commit to run upstream node e2e tests
+
+enable_selinux () {
+    # Make sure SELinux is enabled
+    setenforce 1
```
Comment (TomSweeneyRedHat) [inline scripts/node_e2e_installer:8]: This certainly must be @rhatdan approved coding techniques!  ;^)

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM

File: scripts/node_e2e_installer (hunk 3 above)
Comment (saschagrunert) [reply scripts/node_e2e_installer:8]: I recently wrote my first SELinux module, AMA :grin:

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**